src/Console/AddCommand.php

```diff
@@ -22,25 +22,222 @@
+        $command->addOption('postalCity', null, InputOption::VALUE_OPTIONAL, 'Postal city');
+        $command->addOption('coAddress', null, InputOption::VALUE_OPTIONAL, 'C/o address');
+        $command->addOption('email', null, InputOption::VALUE_OPTIONAL, 'Contact email address');
+        $command->addOption('phone', null, InputOption::VALUE_OPTIONAL, 'Contact phone number');
+        $command->addOption('donationAmount', null, InputOption::VALUE_OPTIONAL, 'Monthly donation amount');
+        $command->addOption('comment', null, InputOption::VALUE_OPTIONAL, 'Comment');
     }
 
     public function execute(InputInterface $input, OutputInterface $output, ContainerInterface $container)
     {
-        // TODO implement...
+        $donorBuilder = $container->get('donor_builder');
+        $donorMapper = $container->get('donor_mapper');
+        $accountFactory = $container->get('account_factory');
+        $idFactory = $container->get('id_factory');
+
+        $this->setPayerNumber(
+            $this->getProperty('payerNumber', 'Unique ID number for donor', '', $input, $output),
+            $donorBuilder
+        );
+        $this->setAccount(
+            $this->getProperty('account', 'Donor account number', '', $input, $output),
+            $donorBuilder,
+            $accountFactory
+        );
+        $this->setId(
+            $this->getProperty('id', 'Donor personal id number or organisation number', '', $input, $output),
+            $donorBuilder,
+            $idFactory
+        );
+        $this->setName(
+            $this->getProperty('name', 'Donor name', '', $input, $output),
+            $donorBuilder
+        );
+        $this->setPostalAddress(
+            [
+                'address1' => $this->getProperty('address1', 'Donor Address line 1', '', $input, $output),
+                'address2' => $this->getProperty('address2', 'Donor Address line 2', '', $input, $output),
+                'postalCode' => $this->getProperty('postalCode', 'Donor Postal code', '', $input, $output),
+                'postalCity' => $this-> getProperty('postalCity', 'Donor Address city', '', $input, $output),
+                'coAddress' => $this->getProperty('coAddress', 'C/o Address', '', $input, $output),
+            ],
+            $donorBuilder
+        );
+        $this->setEmail(
+            $this->getProperty('email', 'Contact email address', '', $input, $output),
+            $donorBuilder
+        );
+        $this->setPhone(
+            $this->getProperty('phone', 'Contact phone number', '', $input, $output),
+            $donorBuilder
+        );
+        $this->setDonationAmount(
+            $this->getProperty('donationAmount', 'Monthly donation amount', '0', $input, $output),
+            $donorBuilder
+        );
+        $this->setComment(
+            $this->getProperty('comment', 'Comment', '', $input, $output),
+            $donorBuilder
+        );
+
+        $donorMapper->save($donorBuilder->buildDonor());
```

Comment:
Here I would also like 

```php
$container->get('event_dispathcer')->dispatch(Events::MANDATE_ADDED_EVENT, new DonorEvent($donor));
```
You will also need to add `MANDATE_ADDED_EVENT` to the `Events` interface if I remember correctly..